src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2447,13 +2564,47 @@ write_deployments_bootswap (OstreeSysroot *self, GPtrArray *new_deployments,
         return glnx_prefix_error (error, "Bootloader write config");
     }
 
-  if (!prepare_new_bootloader_link (self, self->bootversion, new_bootversion, cancellable, error))
+  /* Handle when boot/loader is a symlink (original ostree model) or is a directory (e.g. EFI/vfat)
+   * for recommended systemd type1 BLS
+   */
+  struct stat stbuf;
+  gboolean loader_link = FALSE;
+  gboolean force_type1_semantics = is_bootconfig_type1_semantics (self, cancellable, error);
+  if (!glnx_fstatat_allow_noent (self->sysroot_fd, "boot/loader", &stbuf, AT_SYMLINK_NOFOLLOW,
+                                 error))
     return FALSE;
+  if (errno == ENOENT)
+    loader_link = create_bootself_link (self, cancellable, error);
```

Comment:
Hmm. In an ideal world we wouldn't get here...it's a bit tempting to ask that for the case of a user deploying this *new* version of ostree targeting an ESP we ask them to pre-create the link.

Or...actually another tempting thing to do is check the filesystem type for `boot/loader` and unconditionally require this.

Anyways...for now I think what I'd ask is we continue to properly check errors here...

OK I thought more about this and came up with
https://github.com/ostreedev/ostree/pull/3405
to start.

Then a lot of this code can start checking `sysroot->boot_is_vfat` instead of the return value of `symlinkat`; WDYT?

Reply:
https://github.com/ostreedev/ostree/pull/3405 merged! Let me know what you think!